tests/integration/400.command.dev.test.cjs

```diff
@@ -412,3 +413,69 @@ test('should handle multipart form data when redirecting', async (t) => {
     })
   })
 })
+
+test('should support functions with streaming responses', async (t) => {
+  await withSiteBuilder('site-with-streaming-function', async (builder) => {
+    builder
+      .withPackageJson({ packageJson: { dependencies: { '@netlify/functions': 'latest' } } })
+      .withCommand({ command: ['npm', 'install'] })
+      .withContentFile({
+        content: `
+          const { stream } = require("@netlify/functions");
+
+          class TimerSource {
+            #input;
+            #interval;
+          
+            constructor(input) {
+              this.#input = input;
+            }
+          
+            start(controller) {
+              this.#interval = setInterval(() => {
+                const string = this.#input.shift();
+          
+                if (string === undefined) {
+                  controller.close();
+          
+                  clearInterval(this.#interval);
+          
+                  return;
+                }
+          
+                controller.enqueue(string);
+              }, 50);
+            }
+          
+            cancel() {
+              clearInterval(this.#interval);
+            }
+          }
+          
+          exports.handler = stream(async (event) => ({
+            body: new ReadableStream(new TimerSource(["one", "two", "three"])),
+            headers: {
+              "x-custom-header-1": "value 1"
+            },
+            statusCode: 200,
+          }));
+      `,
+        path: 'netlify/functions/streamer.js',
+      })
+
+    await builder.buildAsync()
+
+    await withDevServer({ cwd: builder.directory }, async (server) => {
+      const chunks = []
+      const response = got.stream(`${server.url}/.netlify/functions/streamer`)
+
+      response.on('data', (chunk) => {
+        chunks.push(chunk.toString())
+      })
+
+      await pause(500)
```

Comment:
let's add an assertion on the arrival times of the chunks, to ensure that local dev isn't buffering the response. i've been bitten by that before.

Reply:
Done in https://github.com/netlify/cli/pull/5725/commits/4bc7c1562fe2a2dfac263b4b45a80af853ac3137.